api/src/main/resources/workbench.yaml

```diff
@@ -797,3 +937,135 @@ definitions:
+        type: string
+    example:
+      val: "12345,1,5"
+
+  # Mirrors notebooks definitions ###############################################
+  ErrorReport:
+    description: ''
+    required:
+      - source
+      - message
+      - causes
+      - stackTrace
+    properties:
+      source:
+        type: string
+        description: service causing error
+      message:
+        type: string
+        description: what went wrong
+      exceptionClass:
+        type: string
+        description: class of exception thrown
+      statusCode:
+        type: integer
+        description: HTTP status code
+      causes:
+        type: array
+        description: errors triggering this one
+        items:
+          $ref: '#/definitions/ErrorReport'
+      stackTrace:
+        type: array
+        description: stack trace
+        items:
+          $ref: '#/definitions/StackTraceElement'
+  StackTraceElement:
+    description: ''
+    required:
+      - className
+      - methodName
+      - fileName
+      - lineNumber
+    properties:
+      className:
+        type: string
+        description: class name
+      methodName:
+        type: string
+        description: method name
+      fileName:
+        type: string
+        description: source file name
+      lineNumber:
+        type: integer
+        description: line number
+  Cluster:
+    description: ''
+    required:
+      - clusterName
+      - googleId
+      - googleProject
+      - googleServiceAccount
+      - googleBucket
+      - operationName
+      - status
+      - createdDate
+      - labels
+    properties:
+      clusterName:
+        type: string
+        description: The user-supplied name for the cluster
+      googleId:
+        type: string
+        description: Google's UUID for the cluster
+      googleProject:
+        type: string
+        description: The Google Project used to create the cluster
+      googleServiceAccount:
+        type: string
+        description: The Google Service Account used to create the cluster
+      googleBucket:
+        type: string
+        description: The Google Bucket associated with the cluster
+      operationName:
+        type: string
+        description: Google's operation ID for the cluster
+      status:
+        type: string
+        description: The current state of the cluster
+      hostIp:
+        type: string
+        description: The IP address of the cluster master node
+      createdDate:
+        type: string
+        description: The date and time the cluster was created, in ISO-8601 format
+      destroyedDate:
+        type: string
+        description: The date and time the cluster was destroyed, in ISO-8601 format
+      labels:
+        type: object
+        description: The labels to be placed on the cluster. Of type Map[String,String]
+  ClusterRequest:
+    description: ''
+    required:
+      - bucketPath
+      - serviceAccount
+      - labels
+    properties:
+      bucketPath:
```

Comment:
The client probably shouldn't be specifying any of this stuff; the server can take care of it. All the client needs to say is "I need a cluster for this workspace".